updated perms in line with verbiage in semgrep app (#805)
returntocorp/semgrep-docs: docs/semgrep-app/getting-started-with-semgrep-app.md

#### Permissions for GitHub

This section explains Semgrep App permissions. Semgrep App requests the following permissions when you log in through GitHub:

<dl>
    <dt>Verify your GitHub identity</dt>
    <dd>Enables Semgrep App to read your GitHub profile data, such as your username.</dd>
    <dt>Know which resources you can access</dt>

>     <dd>Enables Semgrep App to display private repositories you can scan.</dd>

Reviewer: I don't think we have permission to do this when the user login is authorized. I think this permission comes from the GitHub app installation.

Author: Got it, removed it. :)

    <dt>Act on your behalf</dt>

>     <dd>Enables Semgrep App to perform certain tasks <strong>only on resources that you choose to share with Semgrep App</strong>. By default, when you log into Semgrep App for the very first time, no resources are shared. See <a href ="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/authorizing-github-apps">When does a GitHub App act on your behalf?</a></dd>

Reviewer 2: ```suggestion
    <dd>Enables Semgrep App to perform certain tasks <strong>only on resources that you choose to share with Semgrep App</strong>. By default, when you log into Semgrep App for the very first time, no resources are shared. See <a href ="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/authorizing-github-apps">When does a GitHub App act on your behalf?</a> in GitHub documentation.</dd>
```

>     <dd>Enables Semgrep App to perform certain tasks <strong>only on resources that you choose to share with Semgrep App</strong>. By default, when you log into Semgrep App for the very first time, no resources are shared. See <a href ="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/authorizing-github-apps">When does a GitHub App act on your behalf?</a> in GitHub documentation.</dd>

Reviewer: > By default, when you log into Semgrep App for the very first time, no resources are shared.

No resources are shared with the user login authorization, ever, except for viewing username and emails.

And Semgrep never actually acts on the user's behalf. See this screenshot:

<img width="920" alt="CleanShot 2022-12-01 at 12 46 21@2x" src="https://user-images.githubusercontent.com/1060436/205123851-e9a36936-4e43-479b-b0e0-631aa5548e87.png">

If we acted on the user's behalf, the comment would come from @underyx instead of @semgrep-app.

Author: Have worded that we don't use this capability ever. (This would be easier if GitHub didn't call it a permission...)

    <dt>Reading the list of organization members</dt>
    <dd>Enables Semgrep App to determine who can manage your Semgrep organization based on your GitHub organization's members list.</dd>
    <dt>Reading and writing pull requests</dt>
    <dd>Enables Semgrep App to comment about findings on pull requests.</dd>
    <dt>Reading and writing actions</dt>
    <dd>Enables Semgrep App to cancel stuck jobs, rerun jobs, pull logs from jobs, and perform on-demand scanning.</dd>
    <dt>Reading <a href="https://docs.github.com/en/rest/reference/checks">GitHub Checks</a></dt>
    <dd>Facilitates debugging of Semgrep App when configured out of <a href="https://docs.github.com/en/actions">GitHub Actions</a>.</dd>
    <dt>Reading and writing security events</dt>
    <dd>Enables integration with GitHub Advanced Security (for example, to show Semgrep results).</dd>
    <dt>Reading and writing secrets</dt>
    <dd>Enables automatically adding of the Semgrep App Token to your repository secrets when onboarding projects. Note: We cannot read the values of your existing or future secrets (only the names).</dd>
    <dt>Writing (and reading) 2 files</dt>

>     <dd>Enables Semgrep App to configure itself to run in CI by writing to <code>.github/workflows/semgrep.yml</code> and <code>.semgrepignore</code>.</dd>

Reviewer 2: ```suggestion
    <dd>Enables Semgrep App to configure itself to run in CI by writing to <code>.github/workflows/semgrep.yml</code> and <code>.semgrepignore</code> files.</dd>
```

>     <dt>Writing (and reading) workflows</dt>

Reviewer 2: Optional suggestion (feel free to dismiss):
```suggestion
    <dt>Writing and reading workflows</dt>
```

>     <dd>Enables Semgrep App to configure itself to run in CI by writing to <code>.github/workflows/semgrep.yml</code>. GitHub allows writing to files within <code>.github/workflows/</code> only if this permission is granted along with "Writing a single file".</dd>

Reviewer 2: ```suggestion
    <dd>Enables Semgrep App to configure itself to run in CI by writing to <code>.github/workflows/semgrep.yml</code>. GitHub allows writing to files within <code>.github/workflows/</code> directory only if this permission is granted along with "Writing a single file".</dd>
```

>     <dt>Reading PRs, a single file, and workflows</dt>

Reviewer 2: An interesting anacoluthon maybe? - all items should be plural (Reading PRs, single files, and workflows). But I guess this is how GitHub named this permission and we cannot change it. Feel free to resolve and ignore this comment if this cannot be changed.

Reviewer: I'll resolve this discussion and propose a complete change to these two lines

>     <dt>Reading PRs, a single file, and workflows</dt>
>     <dd>Semgrep App only uses write permissions on these resources, but those can only be requested along with read permissions. Semgrep App never accesses your pull request data.</dd>

Reviewer: ```suggestion
    <dt>Writing and reading pull requests</dt>
    <dd>Write permissions allow Semgrep App to leave pull request comments about findings. Read permissions allow Semgrep App to automatically remove findings when the pull request that introduced them is closed without merging.</dd>

```

Author: (Thank you!)
